From 3b91b4f23b7c1461e08de46dc8e37e5ca62c2346 Mon Sep 17 00:00:00 2001
From: Shahed Shaikh <shahedx.shaikh@intel.com>
Date: Tue, 30 Jun 2020 18:09:16 +0800
Subject: [PATCH] DRV_DPM_SW-301: optimize dp_set_bp_attr

Use lightweight function dp_get_netif_subifid() instead of
dp_get_netif_subifid_priv()

Signed-off-by: Shahed Shaikh <shahedx.shaikh@intel.com>
---
 drivers/net/datapath/dpm/datapath_api.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/datapath/dpm/datapath_api.c b/drivers/net/datapath/dpm/datapath_api.c
index f7b1b0fc1852..f387ea47cba5 100644
--- a/drivers/net/datapath/dpm/datapath_api.c
+++ b/drivers/net/datapath/dpm/datapath_api.c
@@ -2634,8 +2634,7 @@ int dp_set_bp_attr(struct dp_bp_attr *conf, u32 flag)
 
 	DP_LIB_LOCK(&dp_lock);
 
-	if (dp_get_netif_subifid_priv(conf->inst, conf->dev, NULL, NULL,
-				      NULL, &subif, 0)) {
+	if (dp_get_netif_subifid(conf->dev, NULL, NULL, NULL, &subif, 0)) {
 		DP_LIB_UNLOCK(&dp_lock);
 		return DP_FAILURE;
 	}
